lib/game_checker.rb

```diff
@@ -0,0 +1,23 @@
+class GameChecker
+  def winner?(board, player)
+    horizontal_winner?(board, player) || vertical_winner?(board, player) || diagonal_winner?(board, player)
+  end
+
+  def tie?(board, player1, player2)
+    board.full? && !winner?(board, player1) && !winner?(board, player2)
+  end
+
+  private
+
+  def horizontal_winner?(board, player)
+    board.find_row(1).all?(player.mark) || board.find_row(2).all?(player.mark) || board.find_row(3).all?(player.mark)
+  end
+
```

Comment:
I like the GameChecker abstraction.  Nice!  A few comments though.

`.all? ` is a method on an array?  In which case, GameChecker knows the data structure of the board.  If you change the data structure of the board, you would also have to change GameChecker.  Also, GameChecker knows the size of the board by passing in 1, 2, and 3.  Do you think GameChecker should know the size of the board?  That information is within the board and for the board to handle.  You could have one board method  such as `board.row_full?` to check if any row is full with the same mark.

Reply:
Yeah this is where I got stuck earlier/ what I meant by dependencies. GameChecker definitely doesn't need to know how big the board is. I like the `board.row_full?` idea!